slimta/bounce.py

```diff
@@ -24,9 +24,11 @@
 import re
 import uuid
 import time
-import cStringIO
+from io import BytesIO
+
 
 from .envelope import Envelope
+from slimta.util.encoders import strict_encode
```

Comment:
I think this file should have a different type of encoding, perhaps `xmlcharref_encode`:

```python
def xmlcharref_encode(unistr):
    return unistr.encode('ascii', 'xmlcharrefreplace')
```

In my testing, most mail clients will correctly render XML character references (e.g. `&#1234;) into the appropriate unicode character, making this a good choice for a bounce message payload.